icechunk/src/session.rs

```diff
@@ -1156,6 +1157,52 @@ async fn node_chunk_iterator<'a>(
     }
 }
 
+// Returns an iterator for chunks in a single split manifest.
+// Each Item contains the _actual_ extents and the ChunkInfo.
+async fn updated_manifest_chunks_iterator<'a>(
+    asset_manager: &'a AssetManager,
+    snapshot_id: &'a SnapshotId,
+    change_set: &'a ChangeSet,
+    node: NodeSnapshot,
+    extent: &ManifestExtents,
+) -> impl Stream<Item = SessionResult<(ManifestExtents, ChunkInfo)>> + 'a {
+    match node.node_data {
+        NodeData::Group => futures::future::Either::Left(futures::stream::empty()),
+        NodeData::Array { manifests, .. } => {
+            // splits on disk; these may not be fully populated
+            // the extents written to disk are the `from`:`to` ranges
+            // of populated chunks
+            let old_splits = ManifestSplits::from_extents(
+                manifests.iter().map(|m| m.extents.clone()).collect(),
+            );
+            // splits created when change_set was modified
+            let new_splits = change_set.splits(&node.id);
+
+            // FIXME: determine if splits are compatible. This is true iff every split
+            // in old_splits is fully contained within exactly one split in new_splits
+            // DISCUSS
+            let compatible = true;
```

Comment:
This bit is messy, because we record the "actual" extents in the snapshot file, not the "theoretical extents". I am pretty sure we can dependably assert that the two extents are compatible.